erpnext/hr/doctype/attendance/attendance.py

```diff
@@ -54,3 +54,54 @@ def validate(self):
 		self.validate_attendance_date()
 		self.validate_duplicate_record()
 		self.check_leave_record()
+
+
+def employee():
+	"""
+	Retun employee id
+
+	:param email: employee company email id
+	"""
+	employee_details, = frappe.db.sql("""
+			SELECT name, employee_name FROM `tabEmployee` WHERE
+			user_id = '{0}' or company_email='{0}'
+			""".format(frappe.session.user), as_dict=1)
+	return employee_details
+
+
+@frappe.whitelist()
+def validate_attendance():
+	"""
+	Check If attendance is already marked
+	"""
+	if frappe.session.user == 'Administrator':
+		return False
+
+	employee_details = employee()
+
+	sql = """
+		SELECT name FROM `tabAttendance` WHERE
+		attendance_date = CURRENT_DATE()
+		and employee = '{}'
+		""".format(employee_details.name)
```

Comment:
Never use .format for SQL queries, leaves it open to injection, again use ORM :)

```
from frappe.utils import nowdate
frappe.get_value('Attendance', dict(employee=employee_details.name, attendance_date=nowdate()))
```